orders/src/main/java/com/orders/service/OrderService.java

```diff
@@ -0,0 +1,316 @@
+          detailOutDto.setFoodItemId(foodItemId);
+          detailOutDto.setFoodItemName(foodItemNameOutDto.getFoodItemName());
+          detailOutDto.setQuantity(quantity);
+          detailOutDto.setPrice(foodItemNameOutDto.getPrice());
+          itemDetails.add(detailOutDto);
+        } else {
+          log.warn("Food item name is null for ID: {}", foodItemId);
+        }
+      } catch (Exception e) {
+        log.error("Error fetching food item name for ID: {} - {}", foodItemId, e.getMessage());
+      }
+    }
+    return itemDetails;
+  }
+
+  /**
+   * Creates a new order based on the provided OrderInDto.
+   * Validates user balance, creates the order, deducts money from the user, and deletes cart entries.
+   *
+   * @param orderInDto the DTO containing order details
+   */
+  public void createOrder(OrderInDto orderInDto) {
+    UserOutDto userOutDto = null;
+    try {
+      userOutDto = userFClient.getUserById(orderInDto.getUserId());
+    } catch (Exception e) {
+      log.error("User not found: {}", e.getMessage());
+      throw new UserNotFound();
+    }
+    if (userOutDto != null && userOutDto.getWalletBalance() < orderInDto.getTotalAmount()) {
+      log.error("Insufficient balance for user ID: {}. Order total: {}", orderInDto.getUserId(), orderInDto.getTotalAmount());
+      throw new InsufficientBalance();
+    }
+
+    Order order = DtoConversion.mapToOrder(orderInDto);
+    order.setOrderDetails(toOrderDetails(orderInDto.getCartIds()));
+
+    AmountInDto amountInDto = new AmountInDto();
+    System.out.println(order.getTotalAmount());
+    amountInDto.setMoney(order.getTotalAmount());
+    // Call Feign client to refund the money
+    userFClient.deductMoneyFromUser(order.getUserId(), amountInDto);
+    orderRepo.save(order);
+    //once the order is placed we delete the corresponding cart entries
+    log.info("Deducting money from user ID: {}. Amount: {}", order.getUserId(), amountInDto.getMoney());
+    for (Integer cartId : orderInDto.getCartIds()) {
+      if (cartRepo.existsById(cartId)) {
+        log.info("Deleted cart entry with ID: {}", cartId);
+        cartRepo.deleteById(cartId);  // Delete if cart exists
+      } else {
+        log.warn("Cart with ID {} does not exist, skipping deletion", cartId);
+      }
+    }
+  }
+  /**
+   * Retrieves the details of an order by its ID.
+   *
+   * @param orderId the ID of the order
+   * @return the OrderOutDto containing order details
+   */
+  public OrderOutDto getOrder(Integer orderId) {
+    Optional<Order> optionalOrder = orderRepo.findById(orderId);
+    if (!optionalOrder.isPresent()) {
+      log.error("Order not found with ID: {}", orderId);
+      throw new OrderNotFound();
+    }
+    Order order = optionalOrder.get();
+    OrderOutDto orderOutDto = DtoConversion.mapToOrderOutDto(order);
+    orderOutDto.setOrderDetails(orderDetailsForOrderOutDto(order.getOrderDetails()));
+    orderOutDto.setAddressName(getAddress(orderOutDto.getAddressId()));
```

Comment:
why we write this logic in controller and why the mapToOrderOutDto is not doing it